packages/common/src/hardhat/tasks/test.js

```diff
@@ -0,0 +1,35 @@
+const path = require("path");
+const { internalTask } = require("hardhat/config");
+const { TASK_TEST_GET_TEST_FILES } = require("hardhat/builtin-tasks/task-names");
+
+const TESTS_DIR = path.resolve(__dirname, "../../../../core/test");
+
+// This overrides a hardhat internal task, which is part of its test task's lifecycle.
```

Comment:
So the idea is to blacklist certain test files for the optimism test network, specifically?

Reply:
Other way around: blacklist `optimism/` tests for other networks. But now that I think about it, a test `whitelist` might make more sense so we can cherry pick which tests to run in the Optimism environment.

Reply:
Or, using the `blacklist` to block non-OVM compatible tests. I'll go with this solution as you suggested.